### PR TITLE
Added Genome related events

### PR DESCRIPTION
**Pull request by vedant-shroff**
added genome related events to be handled by the GenomeAuthoritySystem if Genomes is enabled. 

*To-do before merging*
- [x] Write modifications for non sustainable bushes to inherit GenomeComponent and not throw NPE
- [x] Major Code Cleanup
- [x] Remove Genome Dependency. That was only for testing purposes
- [x] Add JavaDoc in the new Handlers and events

### CODE REVIEW

File: module.txt [hunk 1]
```diff
@@ -8,7 +8,6 @@
         { "id": "CoreAssets", "minVersion": "2.0.1" },
         { "id": "Inventory", "minVersion": "1.1.0" },
         { "id": "ModuleTestingEnvironment", "minVersion": "0.2.0", "optional": true }
-    ],
```
Comment (syntaxi) [inline module.txt:11]: I think this is a mistake?

File: src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java [hunk 2]
```diff
@@ -168,23 +174,27 @@ public void onCheatGrowth(ActivateEvent event, EntityRef item, CheatGrowthCompon
...
      * @param bushComponent the definition of the bush being grown
-     * @param stages        the number of stages to grow; negative values represent un-growth
+     * @param stages the number of stages to grow; negative values represent un-growth
      */
-    private void doBushGrowth(Vector3i position, BushDefinitionComponent bushComponent, int stages) {
+    private void doBushGrowth(Vector3i position, EntityRef bush, BushDefinitionComponent bushComponent, int stages) {
         if (!isInLastStage(bushComponent)
                 // allow negative growth from the last stage
                 || stages < 0) {
+            bush.send(new AddGenomeRetention());
```
Comment (syntaxi) [inline src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java:183]: This feels like a bit clunky and weird to be doing through events tbh. That being said, given the state of optional dependencies this possibly is the best way to do this.

File: src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java [hunk 3]
```diff
@@ -168,23 +174,27 @@ public void onCheatGrowth(ActivateEvent event, EntityRef item, CheatGrowthCompon
...
+            newBush.send(new AddGenomeRetention());
             newBush.addOrSaveComponent(bushComponent);
 
             if (stage.getValue().maxTime > 0 && stage.getValue().minTime > 0) {
                 resetDelay(newBush,
                         stage.getValue().minTime,
                         stage.getValue().maxTime);
             }
+
+            bush.send(new TransferGenomeEvent(newBush));
```
Comment (syntaxi) [inline src/main/java/org/terasology/simpleFarming/systems/BushAuthoritySystem.java:197]: This event by contrast seems a bit weird than that there _should_ be a better way to do it.

**Review by syntaxi — COMMENTED** [review]
Overall seems fine,
I am a little bit wary of the `TransferGenomeEvent`
I've put more details about this and more in the review for
Terasology/EdibleFlora#5

**Review by e-aakash — APPROVED** [review]

**Pull request merged**